### PR TITLE
Clippy

### PR DESCRIPTION
**Pull request by dtwood**

### CODE REVIEW

File: grease-socket/src/lib.rs [hunk 1]
```diff
@@ -27,8 +32,8 @@ extern crate rand;
 use std::collections::{HashMap, VecDeque};
 use std::convert::From;
 use std::fmt;
-use std::io::prelude::*;
 use std::io;
+use std::io::prelude::*;
```
Comment (jonathanpallant) [inline grease-socket/src/lib.rs:36]: When I build in 1.26, it tells me this io::prelude import is complete redundant.

Comment (dtwood) [reply grease-socket/src/lib.rs:36]: I'm surprised by that - builds ok for me, and if I remove it, this happens:
```diff
diff --git a/grease-socket/src/lib.rs b/grease-socket/src/lib.rs
index 673cde7..21dbb05 100644
--- a/grease-socket/src/lib.rs
+++ b/grease-socket/src/lib.rs
@@ -33,7 +33,6 @@ use std::collections::{HashMap, VecDeque};
 use std::convert::From;
 use std::fmt;
 use std::io;
-use std::io::prelude::*;
 use std::net;
 use std::thread;
```
```
PS C:\Users\david\Devel\grease\grease-socket> rustc --version
rustc 1.27.0-nightly (565235ee7 2018-05-07)
PS C:\Users\david\Devel\grease\grease-socket> cargo build
   Compiling grease-socket v0.2.0 (file:///C:/Users/david/Devel/grease/grease-socket)
error[E0599]: no method named `write` found for type `mio::net::TcpStream` in the current scope
   --> grease-socket\src\lib.rs:476:24
    |
476 |             match cs.connection.write(&pw.data[pw.sent..]) {
    |                                 ^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope, perhaps add a `use` for it:
    |
32  | use std::io::Write;
    |

error[E0599]: no method named `read` found for type `mio::net::TcpStream` in the current scope
   --> grease-socket\src\lib.rs:526:25
    |
526 |                 match cs.connection.read(buffer.as_mut_slice()) {
    |                                     ^^^^
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope, perhaps add a `use` for it:
    |
32  | use std::io::Read;
    |

error[E0599]: no method named `write` found for type `mio::net::TcpStream` in the current scope
   --> grease-socket\src\lib.rs:666:25
    |
666 |                 match cs.connection.write(&req_send.data) {
    |                                     ^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope, perhaps add a `use` for it:
    |
32  | use std::io::Write;
    |

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0599`.
error: Could not compile `grease-socket`.

To learn more, run the command again with --verbose.
```

and under WSL:
```
root@david-laptop:/mnt/c/Users/david/Devel/grease/grease-socket# rustc --version
rustc 1.26.0 (a77568041 2018-05-07)
root@david-laptop:/mnt/c/Users/david/Devel/grease/grease-socket# cargo build
   Compiling grease-socket v0.2.0 (file:///mnt/c/Users/david/Devel/grease/grease-socket)
error[E0599]: no method named `write` found for type `mio::net::TcpStream` in the current scope
   --> grease-socket/src/lib.rs:476:24
    |
476 |             match cs.connection.write(&pw.data[pw.sent..]) {
    |                                 ^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope, perhaps add a `use` for it:
    |
32  | use std::io::Write;
    |

error[E0599]: no method named `read` found for type `mio::net::TcpStream` in the current scope
   --> grease-socket/src/lib.rs:526:25
    |
526 |                 match cs.connection.read(buffer.as_mut_slice()) {
    |                                     ^^^^
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope, perhaps add a `use` for it:
    |
32  | use std::io::Read;
    |

error[E0599]: no method named `write` found for type `mio::net::TcpStream` in the current scope
   --> grease-socket/src/lib.rs:666:25
    |
666 |                 match cs.connection.write(&req_send.data) {
    |                                     ^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope, perhaps add a `use` for it:
    |
32  | use std::io::Write;
    |

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0599`.
error: Could not compile `grease-socket`.

To learn more, run the command again with --verbose.
```

Comment (dtwood) [reply grease-socket/src/lib.rs:36]: Ah, you mean line 788 in the tests

File: grease-socket/src/lib.rs [hunk 2]
```diff
@@ -779,22 +773,22 @@ impl fmt::Debug for ReqSend {
...
 		SocketError::IOError(e.kind())
 	}
 }
 
 #[cfg(test)]
 mod test {
+	use super::*;
+	use grease::prelude::*;
+	use rand::Rng;
 	use std::io::prelude::*;
```
Comment (dtwood) [inline grease-socket/src/lib.rs]: ^ this one

File: grease-socket/src/lib.rs (hunk 1 above)
Comment (dtwood) [reply grease-socket/src/lib.rs:36]: Fixed now

Comment (jonathanpallant) [reply grease-socket/src/lib.rs:36]: Oops, yes I did!

**Comment by jonathanpallant** [comment]
I <3 clippy. Thanks for the PR!

**Pull request merged**